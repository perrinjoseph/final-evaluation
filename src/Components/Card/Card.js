import React from "react";
import Button from "../Button/Button";

function Card({ title, body, image, project, bg }) {
  switch (project) {
    case true:
      return (
        <article
          style={{
            background: `linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
        url(${bg})`,
          }}
          className="project--card"
        >
          <h3 className="main__content">{title}</h3>
          <p className="content__heading">{body}</p>
          <Button content={"More"}/>
        </article>
      );
    default:
      return (
        <div className="cardglobal">
          <img className="cardglobal__img" src={image}></img>
          <section>
            <h3>{title}</h3>
            <p>{body}</p>
          </section>
        </div>
      );
  }
}

export default Card;
