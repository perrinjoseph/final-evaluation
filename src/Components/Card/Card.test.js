import { render } from "@testing-library/react";
import Card from "./Card";
describe("Card", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<Card>Snapshot Card</Card>);
    expect(asFragment()).toMatchSnapshot();
  });
});
