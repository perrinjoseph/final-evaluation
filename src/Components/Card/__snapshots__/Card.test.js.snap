// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="cardglobal"
  >
    <img
      class="cardglobal__img"
    />
    <section>
      <h3 />
      <p />
    </section>
  </div>
</DocumentFragment>
`;
